File: Structural/Decorator/decorator.py
class Component():
    def operation(self) -> str: pass

class ConcreteComponent(Component):
    def operation(self) -> str:
        return "ConcreteComponent"

class Decorator(Component):
    _component: Component = None
    def __init__(self, component: Component):
        self._component = component
    def operation(self) -> str:
        return self._component.operation()

class ConcreteDecoratorA(Decorator):
    def operation(self) -> str:
        return f"ConcreteDecoratorA({self._component.operation()})"

File: Structural/Decorator/decorator.cs
public abstract class Component { public abstract string Operation(); }

class ConcreteComponent : Component {
    public override string Operation() => "ConcreteComponent";
}

abstract class Decorator : Component {
    protected Component _component;
    public Decorator(Component component) => _component = component;
}

class ConcreteDecoratorA : Decorator {
    public ConcreteDecoratorA(Component c) : base(c) {}
    public override string Operation() => $"ConcreteDecoratorA({_component.Operation()})";
}